tests/validation/tests/v3_api/common.py

```diff
@@ -2735,3 +2735,113 @@ def get_node_details(cluster, client):
+    ca = res["data"]["ca.crt"]
+    token = res["data"]["token"]
+    token = base64.b64decode(token).decode()
+
+    # get the server ip, which is the control plane's ip
+    ip = os.environ.get('RANCHER_CONTROL_PLANE_IP', "None")
+    server = "https://" + ip + ":6443"
+
+    config = """
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - name: test-cluster
+      cluster:
+        server: {server}
+        certificate-authority-data: {ca}
+    contexts:
+    - name: default-context
+      context:
+        cluster: test-cluster
+        namespace: default
+        user: test-user
+    current-context: default-context
+    users:
+    - name: test-user
+      user:
+        token: {token}
+    """
+    config = config.format(server=server, ca=ca, token=token)
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               name + ".yaml")
+    with open(config_file, "w") as file:
+        file.write(config)
+
+    return name
+
+
+def test_reader(file_path=None):
+    """
+    This method generates test cases from an input file and return the result
+    that can be used to parametrize pytest cases
+    :param file_path: the path to the JSON file for test cases
+    :return: a list of tuples of (cluster_role, command, authorization)
+    """
+
+    if file_path is None:
+        pytest.fail("no file is provided")
+    with open(file_path) as reader:
+        test_cases = json.loads(reader.read().replace("{resource_root}",
+                                                      DATA_SUBDIR))
+    output = []
+    for cluster_role, checks in test_cases.items():
+        # create a service account for each role
+        name = create_service_account_configfile()
+        for command in checks["should_pass"]:
+            output.append((cluster_role, command, True, name))
+        for command in checks["should_fail"]:
+            output.append((cluster_role, command, False, name))
+
+    return output
+
+
+def validate_cluster_role_rbac(cluster_role, command, authorization, name):
+    """
+     This methods creates a new service account to validate the permissions
+     both before and after creating the cluster role binding between the
+     service account and the cluster role
+    :param cluster_role:  the cluster role
+    :param command: the kubectl command to run
+    :param authorization: if the service account has the permission: True/False
+    """
+
+    # create the cluster role binding if it does not exist
+    result = execute_kubectl_cmd(cmd='get clusterrolebinding {}'.format(name),
+                                 json_out=False,
+                                 stderr=True).decode('utf_8')
+    if "NotFound" in result:
+        cmd = "create clusterrolebinding {} " \
+              "--clusterrole {} " \
+              "--serviceaccount {}".format(name, cluster_role,
+                                           "default:" + name)
+        execute_kubectl_cmd(cmd, json_out=False)
```

Comment:
@jiaqiluo Why cant we do the clusterrolebindings along with service account creation ?

Reply:
you are right. 
This is moved out.